### PR TITLE
list.reduce implemented via recursion, some tests disabled

### PR DESCRIPTION
**Pull request by EugeneDar**

### CODE REVIEW

**Comment by EugeneDar** [comment]
@yegor256 @Graur check please.

**Comment by EugeneDar** [comment]
It is important to note that due to the recursive implementation of list.reducei, we had to temporarily disable (wrap to 'nop') some methods in multimap.eo. Because otherwise they would overflow the stack. @yegor256 is this ok?

File: eo-runtime/src/test/eo/org/eolang/collections/list-tests.eo [hunk 1]
```diff
@@ -366,3 +366,19 @@
         list
           args.
             b 1 2 "fail"
+
+[] > recursion-falls-test
+  reducei2. > res!
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/list-tests.eo:371]: @EugeneDar This is a typo?

File: eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo [hunk 2]
```diff
@@ -48,18 +48,14 @@
...
-  assert-that > @
-    * (r1.at 0) (r2.length) (r3.at 0) (r4.at 0) (rabc.at 0)
-    $.equal-to
-      list
-        * "a" 0 5 5 "cba"
+  nop > @
+    map * > init-mp!
+    (init-mp.with 1 "a").with "abc" "cba" > mp!
+    assert-that
+      * ((mp.found 1).at 0) ((mp.found "abc").at 0)
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo:55]: @EugeneDar Can we simplify this test?

File: eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo [hunk 3]
```diff
@@ -74,14 +70,15 @@
...
-  (((init-mp.with 1 "a").with 3 "c").with 2 "b").with 4 "d" > mp!
-  (mp.without 2).without 3 > mp2!
-  assert-that > @
-    * ((mp.found 1).length) ((mp.found 2).length) ((mp2.found 3).length) ((mp2.found 4).length)
-    $.equal-to
-      list
-        * 1 1 0 1
+  nop > @
+    map * > init-mp!
+    (((init-mp.with 1 "a").with 3 "c").with 2 "b").with 4 "d" > mp!
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo:75]: @EugeneDar Same here, can we simplify this test?

File: eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo [hunk 4]
```diff
@@ -94,10 +91,9 @@
...
 
 [] > map-large-key-test
-  map * > init-mp!
-  init-mp.with "key123" 42 > mp!
-  assert-that > @
-    * (mp.size) ((mp.found "key123").at 0)
-    $.equal-to
-      list
-        * 1 42
+  nop > @
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/map-tests.eo:94]: @EugeneDar I think we can get rid of this test, as part of perfomance test

File: eo-runtime/src/test/eo/org/eolang/collections/multimap-tests.eo [hunk 5]
```diff
@@ -144,18 +106,14 @@
...
-  assert-that > @
-    * (r1.at 0) (r2.length) (r3.at 0) (r4.at 0) (rabc.at 0)
-    $.equal-to
-      list
-        * "a" 0 5 5 "cba"
+  nop > @
+    multimap * > init-mp!
+    init-mp.with 1 "a" > mp!
+    assert-that
+      * ((mp.found 1).at 0) ((mp.found "abc").length)
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/multimap-tests.eo:113]: @EugeneDar Same here, can we simplify this test?

File: eo-runtime/src/test/eo/org/eolang/collections/multimap-tests.eo [hunk 6]
```diff
@@ -196,10 +154,11 @@
     $.equal-to 0
 
 [] > multimap-large-key-test
-  (multimap *).with "key123" 42 > mp!
-  assert-that > @
-    (mp.found "key123").length
-    $.equal-to 1
+  nop > @
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/collections/multimap-tests.eo:157]: @EugeneDar I think we can get rid of this test, as part of perfomance test

**Review by Graur — CHANGES_REQUESTED** [review]
@EugeneDar Good one! Just a few comments below

File: eo-runtime/src/test/eo/org/eolang/collections/list-tests.eo (hunk 1 above)
Comment (EugeneDar) [reply eo-runtime/src/test/eo/org/eolang/collections/list-tests.eo:371]: @Graur you're right, I'll fix it.

**Comment by EugeneDar** [comment]
@Graur please check again.

**Comment by EugeneDar** [comment]
@rultor merge

**Review by Graur — APPROVED** [review]

**Comment by Graur** [comment]
@yegor256 Please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Comment by EugeneDar** [comment]
@rultor hello

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**